### PR TITLE
Global overlay fix

### PR DESCRIPTION
**Pull request by steppy452**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The `open` prop passed to the overlay was only used as a default value of sorts to enable the overlay. Then the state changes within the internal component took over and it was impossible to cancel closing the overlay once `onClose` had been triggered (ie, you could not add a confirmation prompt).

Issue Number: N/A

## What is the new behavior?
`open` prop is synced with internal component state. `onClose` prop is called from the internal component instead of changing the internal `open` state.

## Does this PR introduce a breaking change?
```
[x] Yes - possibly
[ ] No
```

If `onClose` prop was not passed in previously, the component incorrectly handled it before. Now, if onClose isn't passed in, the overlay will not be able to close. See story updates.

@amcdnl ☝️should `onClose` be a required prop instead?

### CODE REVIEW

File: src/Overlay/GlobalOverlay.tsx [hunk 1]
```diff
@@ -20,24 +20,12 @@ export const GlobalOverlay: FC<GlobalOverlayProps> = ({
...
-  } = useGlobalOverlay(rest);
-  const mounted = useRef<boolean>(false);
+  } = useGlobalOverlay({ onClose, ...rest });
   const renderFn = children || render;
 
   useEffect(() => {
     setOpen(open);
   }, [open]);
 
-  useEffect(() => {
```
Comment (amcdnl) [inline src/Overlay/GlobalOverlay.tsx:32]: Why was this removed?

File: src/Overlay/Overlay.story.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ storiesOf('Utilities|Overlay/Portal', module)
     <div
       style={{
         width: 300,
-        height: 300
+        height: 300,
```
Comment (amcdnl) [inline src/Overlay/Overlay.story.tsx:17]: Can you update the prettier config to no trailling `,`

File: src/Overlay/GlobalOverlay.tsx (hunk 1 above)
Comment (steppy452) [reply src/Overlay/GlobalOverlay.tsx:32]: I don't think it's necessary since `onClose` is being called within `useGlobalOverlay`. This was causing `onClose` to be called twice
